File: fms_backend/fms-user/src/main/java/com/uth/fms/user/services/AuthService.java
package com.uth.fms.user.services;

import com.uth.fms.user.dto.reponse.LoginResponse;
import com.uth.fms.user.dto.request.LoginRequest;
import com.uth.fms.user.dto.request.RegisterRequest;

public interface AuthService {
    
    LoginResponse login(LoginRequest request);
    String register(RegisterRequest request);

    /**
     * Làm mới accessToken đã hết hạn bằng refreshToken
     * @param refreshToken
     * @return 1 accessToken dạng String
     */
    String refreshAccessToken(String refreshToken);

}

File: fms_frontend/src/services/authService.ts
import type {LoginRequest} from "../types/api/request/auth.ts";
import axiosInstance from "../api/axiosInstance.ts";
import type {LoginResponse} from "../types/api/response/auth.ts";

export const authService = {
    /**
     * API Đăng nhập hệ thống FMS
     * @param payload Chứa thông tin tài khoản gồm username và password
     */
    login: async (payload: LoginRequest) => {
        const response = await axiosInstance.post<{ data: LoginResponse }>(
            '/auth/login',
            payload
        );
        return response.data;
    },

    /**
     * API Đăng xuất hệ thống (Vô hiệu hóa token trên Redis)
     */
    logout: async () => {
        const response = await axiosInstance.post('/auth/logout');
        return response.data;
    }
}